Modules/readline.c

```diff
@@ -601,60 +678,75 @@ py_add_history(PyObject *self, PyObject *string)
     Py_RETURN_NONE;
 }
 
-PyDoc_STRVAR(doc_add_history,
-"add_history(string) -> None\n\
-add an item to the history buffer");
-
 static int should_auto_add_history = 1;
 
 /* Enable or disable automatic history */
 
+/*[clinic input]
+readline.set_auto_history
+
+    enabled as _should_auto_add_history: bool
+    /
+
+Enables or disables automatic history.
+[clinic start generated code]*/
+
 static PyObject *
-py_set_auto_history(PyObject *self, PyObject *args)
+readline_set_auto_history_impl(PyObject *module,
+                               int _should_auto_add_history)
+/*[clinic end generated code: output=619c6968246fd82b input=3d413073a1a03355]*/
 {
-    if (!PyArg_ParseTuple(args, "p:set_auto_history",
-                          &should_auto_add_history)) {
-        return NULL;
-    }
+    should_auto_add_history = _should_auto_add_history;
     Py_RETURN_NONE;
 }
 
-PyDoc_STRVAR(doc_set_auto_history,
-"set_auto_history(enabled) -> None\n\
-Enables or disables automatic history.");
-
 
 /* Get the tab-completion word-delimiters that readline uses */
 
+/*[clinic input]
+readline.get_completer_delims
+
+get the word delimiters for completion
+[clinic start generated code]*/
+
 static PyObject *
-get_completer_delims(PyObject *self, PyObject *noarg)
+readline_get_completer_delims_impl(PyObject *module)
+/*[clinic end generated code: output=6b060280fa68ef43 input=51134d5723987d99]*/
 {
     return decode(rl_completer_word_break_characters);
 }
 
-PyDoc_STRVAR(doc_get_completer_delims,
-"get_completer_delims() -> string\n\
-get the word delimiters for completion");
+/* Set the completer function */
 
+/*[clinic input]
+readline.set_completer
 
-/* Set the completer function */
+    function: object = None
+    /
+
+Set or remove the completer function.
+
+The function is called as function(text, state),
+for state in 0, 1, 2, ..., until it returns a non-string.
+It should return the next possible completion starting with 'text'.
+[clinic start generated code]*/
 
 static PyObject *
-set_completer(PyObject *self, PyObject *args)
+readline_set_completer_impl(PyObject *module, PyObject *function)
+/*[clinic end generated code: output=171a2a60f81d3204 input=51e81e13118eb877]*/
 {
-    return set_hook("completer", &readlinestate_global->completer, args);
+    return set_hook("completer", &readlinestate_global->completer, function);
 }
 
-PyDoc_STRVAR(doc_set_completer,
-"set_completer([function]) -> None\n\
-Set or remove the completer function.\n\
-The function is called as function(text, state),\n\
-for state in 0, 1, 2, ..., until it returns a non-string.\n\
-It should return the next possible completion starting with 'text'.");
+/*[clinic input]
+readline.get_completer
 
+Returns current completer function.
```

Comment:
Maybe use "get" for uniformity with other functions?

At least "return" is more preferable than "returns".